src/commands/add-task.stanza

```diff
@@ -0,0 +1,282 @@
+defpackage slm/commands/add-task:
+  import core
+  import collections
+  import arg-parser
+
+  import semver
+  import maybe-utils
+
+  import slm/flags
+  import slm/toml
+  import slm/dep-types
+  import slm/errors
+  import slm/file-utils
+  import slm/logging
+  import slm/platforms
+
+
+defn cli-add-task (cmd-args:CommandArgs) -> False :
+
+  val cfg-path = get?(cmd-args, "cfg", SLM_TOML_NAME)
+
+  val name = get(cmd-args, "name")
+  val command = get(cmd-args, "command")
+  val platform? = to-maybe $ get?(cmd-args, "platform", false)
+  val args? = to-maybe $ get?(cmd-args, "args", false)
+  val working-dir? = to-maybe $ get?(cmd-args, "working-dir", false)
+  val env-vars = get?(cmd-args, "env-vars", [])
+
+  val force = get?(cmd-args, "force", false)
+  val dry-run? = get?(cmd-args, "dry-run", false)
+
+  debug("Add-Task Parameters:")
+  val paramTable = [
+    ["name", name]
+    ["command", command]
+    ["platform", platform?]
+    ["args", args?]
+    ["working-dir?", args?]
+    ["env-vars", env-vars]
+    ["force", force]
+    ["dry-run", dry-run?]
+  ]
+  for kvp in paramTable do:
+    debug("\t%_ = %_" % kvp)
+
+  val cfg = parse-project-toml(cfg-path, env-sub-enable = false)
+
+  val env-vars* = parse-env-vars-flag(env-vars)
+
+  val cfg* = add-task-dependency(
+    cfg,
+    name, command,
+    platform? = platform?,
+    args? = args?,
+    working-dir? = working-dir?,
+    env-vars = env-vars*,
+    force = force
+    )
+
+  debug("-- Start New File Write --")
+
+  if dry-run?:
+    val o = current-output-stream()
+    write(o, cfg*)
+  else:
+    within f = open(cfg-path, false):
+      write(f, cfg*)
+
+  debug("-- New File Write Complete --")
+
+defn remove-existing-task (existing:TaskDependency, name:String, platform?:Maybe<String>, force:True|False) -> TaskDependency :
+  match(platform?):
+    (x:None):
+      if task?(existing) is-not None and not force:
+        throw $ ValueError("Task '%_' Already Exists" % [name])
+
+      sub-task?(existing, None())
+    (x:One<String>):
+      val platform = value(x)
+      val plat-tasks = to-hashtable<String, Task> $ platform-tasks(existing)
+      val plat-task = to-maybe $ get?{_, platform, false} $ plat-tasks
```

Comment:
Can we remove the task more directly? Isn't `platform-tasks(existing)` already a `HashTable` that you can mutate?

Reply:
Ya - I had originally implemented as a `Tuple<KeyValue>` and so I couldn't remove it directly. I think I didn't update after making the change to `HashTable`. I think I was also trying to keep it as immutable as possible but I'm not sure that really matters in this case.